Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectBuilder/MSBuildProjectBuilder.cs

```diff
@@ -189,6 +199,27 @@ void DisplayProgress(string status = "", float progress = 0)
                             {
                                 cancellationTokenSource.Cancel();
                             }
+
+                            if (adoAuthenticationMatch?.Success == true)
```

Comment:
>?.Success == true [](start = 54, length = 17)

Technically you're doing the Success check twice which shouldn't really be needed all within a single private function.

Reply:
Addressed with some minor changes.